gems/aws-sdk-core/spec/aws/instance_profile_credentials_spec.rb

```diff
@@ -163,42 +150,40 @@ module Aws
       let(:expiration) { (Time.now + 3600).utc.iso8601 }
 
       before do
-        stub_request(:put, ipv4_endpoint + token_path)
-          .to_return(status: 200, body: "my-token\n", headers: { 'x-aws-ec2-metadata-token-ttl-seconds' => '21600' })
+        stub_request(:put, ipv4_endpoint + token_path).to_return(status: 200, body: "my-token\n")
       end
 
       it 'Test IMDS credentials provider returns valid credentials with account ID' do
-        stub_request(:get, "#{ipv4_endpoint}#{extended_path}")
-          .to_return(status: 200, body: "my-profile-0001\n")
         resp1 = {
-          "Code": "Success",
-          "LastUpdated": "2025-03-12T20:53:17.832308Z",
-          "Type": "AWS-HMAC",
-          "AccessKeyId": "ASIAIOSFODNN7EXAMPLE",
-          "SecretAccessKey": "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY",
-          "Token": "AQoEXAMPLEH4aoAH0gNCAPyJxz4BlCFFxWNE1OPTgk5TthT+FvwqnKw...(truncated)",
+          "Code": 'Success',
```

Comment:
I didn't look at the contents of these jsons (I copied them from the test json file) but we can consider DRYing these up with a method that returns json, and maybe as a parameter it takes expiration or account id etc (or you can merge against defaults). Otherwise it's fine as big blobs.